### PR TITLE
persist: switch the remaining Vars configs to the new dyn config system

### PR DESCRIPTION
**Pull request by danhhz**
Added two new `ConfigType`s: `u32` and `Duration`.

It turns out that a few DynamicConfigs were never actually hooked up to
vars for dynamic configuration. Leaving these for a followup, to try to
break this up a tiny bit. Anyway, we may want to make everything in
PersistConfig dynamically configurable now that it's easy, and these
would fit pretty naturally in that category.

Mostly mechanical modulo making sure that some of the configs are still
available externally to be overridden by tests outside of
persist-client.

### Motivation

  * This PR adds a feature that has not yet been specified.

### Tips for reviewer

- Pulled what I thought might be a controversial swap of how the untrimmable columns work into the first commit, so it's easier to see.
- I'm going to double check that I didn't mess up anything mechanical (e.g. change name of config, change default of config, get the value of the wrong config) but it'd be great to have a second set of eyes do the same audit.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by mjibson — COMMENTED** [review]
Adapter parts lgtm.

File: src/persist-client/src/rpc.rs [hunk 1]
```diff
@@ -1014,10 +1031,10 @@ impl proto_persist_pub_sub_server::ProtoPersistPubSub for PersistGrpcPubSubServe
                         Some(proto_pub_sub_message::Message::PushDiff(req)) => {
                             let shard_id = req.shard_id.parse().expect("valid shard id");
                             let diff = VersionedData {
-                                seqno: req.seqno.into_rust().expect("WIP"),
+                                seqno: req.seqno.into_rust().expect("valid seqno"),
```
Comment (bkirwi) [inline src/persist-client/src/rpc.rs:1034]: 🙇

**Review by bkirwi — APPROVED** [review]
I'm a bit on the fence about whether it's worthwhile supporting the `u32` type, since AFAICT it's only used ~once. But either way this is great! Will be much easier to manage.

**Comment by danhhz** [comment]
> I'm a bit on the fence about whether it's worthwhile supporting the `u32` type, since AFAICT it's only used ~once. But either way this is great! Will be much easier to manage.

Good point! Looks like there are a good number of `ServerVar<32>` in vars.rs (24 from a quick grep). Hopefully some of those get switched once we pull this out into a lib, which would make it feel better?

**Comment by bkirwi** [comment]
Yep for sure!

**Comment by danhhz** [comment]
TFTR!

**Pull request merged**